lib/marc_access_facet_processor.rb

```diff
@@ -43,4 +43,12 @@ def resolve_library_code(field, library)
 
     'On Order'
   end
+
+  def hathi_access?(context)
+    hathitrust_struct = context.output_hash&.dig('hathitrust_struct')
+    hathitrust_etas = context.settings['hathi_etas']
+    return false unless hathitrust_struct
+
+    hathitrust_etas || (JSON.parse(hathitrust_struct&.first)['access'] == 'allow' && !hathitrust_etas)
```

Comment:
```suggestion
    hathitrust_etas || JSON.parse(hathitrust_struct&.first)['access'] == 'allow'
```

Seems redundant to check `hathitrust_etas` twice. If `hathitrust_etas` is true, the part of this statement after `||` never gets evaluated. If it is false, the part after `||` will mean `!hathitrust_etas` always evaluates to true. Right?

Reply:
yes. right 👍